# Clinical Risk Analysis: Diabetes and Hypertension in 1000 Synthetic Patients

This notebook explores a **synthetic** patient dataset (1,000 rows) using NumPy to practice array operations: summary statistics, boolean masking, and combining conditions.

> **Note:** This dataset is synthetic/simulated and does not represent real patients or real clinical findings. It is for practicing data analysis.

## 1. Setup

In [ ]:
import numpy as np
import pandas as pd

## 2. Load the data

Reads the CSV into a pandas DataFrame, then pulls out each column as a NumPy array for analysis.

In [ ]:
df = pd.read_csv("synthetic_diabetes_hypertension_patient_dataset.csv")

age = df["age_years"].to_numpy()
bmi = df["bmi_kg_m2"].to_numpy()
sbp = df["systolic_bp_mmHg"].to_numpy()
dbp = df["diastolic_bp_mmHg"].to_numpy()
fastglc = df["fasting_glucose_mg_dL"].to_numpy()
a1c = df["hba1c_percent"].to_numpy()
creatinine = df["creatinine_mg_dL"].to_numpy()
cholesterol_total = df["total_cholesterol_mmol_L"].to_numpy()
hdl = df["hdl_mmol_L"].to_numpy()
diabetes = df["diabetes"].to_numpy()
htn = df["hypertension"].to_numpy()

df.shape

(1000, 12)

## 3. Demographics: age, BMI, blood pressure

`np.nanmean` / `np.nanmax` are used for columns that contain missing values (`bmi_kg_m2`, `hba1c_percent`), so a few `NaN` entries don't turn the whole result into `NaN`.

In [ ]:
mean_age = np.mean(age)
median_age = np.median(age)
mean_bmi = np.nanmean(bmi)
mean_sbp = np.mean(sbp)
std_sbp = np.std(sbp)

print(f"Mean age: {mean_age}")
print(f"Median age: {median_age}")
print(f"Mean BMI: {mean_bmi} kg/m2")
print(f"Mean systolic blood pressure is: {mean_sbp} mmHg")
print(f"The standard deviation of systolic blood pressure is: {std_sbp}")

Mean age: 51.622
Median age: 52.0
Mean BMI: 27.36237113402062 kg/m2
Mean systolic blood pressure is: 139.764 mmHg
The standard deviation of systolic blood pressure is: 18.184177297859808


## 4. Clinical extremes: HbA1c and fasting glucose

In [ ]:
highest_HbA1c = np.nanmax(a1c)
minimum_fastglc = np.min(fastglc)

print(f"Highest HbA1c is: {highest_HbA1c}")
print(f"Lowest fasting glucose is: {minimum_fastglc} mg/dl")

Highest HbA1c is: 9.1
Lowest fasting glucose is: 55.0 mg/dl


## 5. Dataset size

In [ ]:
total_patients = len(df)

print(f"Number of patients in the dataset: {total_patients}")

Number of patients in the dataset: 1000


## 6. Diabetes prevalence

`diabetes == "Yes"` compares every entry to `"Yes"` and returns a boolean array (one True/False per patient). Since `True` counts as `1` and `False` as `0`, `np.sum(...)` on that array counts how many patients have diabetes.

In [ ]:
diabetic = diabetes == "Yes"
non_diabetic_mask = diabetes == "No"

diabetes_count = np.sum(diabetic)
non_diabetic_count = np.sum(non_diabetic_mask)
diabetes_percentage = (diabetes_count / len(diabetes)) * 100

print(f"Number of patients with diabetes: {diabetes_count}")
print(f"Number of patients without diabetes: {non_diabetic_count}")
print(f"The prevalence of people having diabetes is: {diabetes_percentage}%")

Number of patients with diabetes: 151
Number of patients without diabetes: 849
The prevalence of people having diabetes is: 15.1%


## 7. Hypertension prevalence

Same boolean-mask pattern, applied to the `hypertension` column.

In [ ]:
hypertensive = htn == "Yes"
non_hypertensive_mask = htn == "No"

hypertension_count = np.sum(hypertensive)
non_hypertension_count = np.sum(non_hypertensive_mask)
hypertension_percentage = (hypertension_count / len(htn)) * 100

print(f"The number of people who are hypertensive are: {hypertension_count}")
print(f"The number of patients who are non-hypertensive are: {non_hypertension_count}")
print(f"The percentage of patients with hypertension is: {hypertension_percentage}%")

The number of people who are hypertensive are: 500
The number of patients who are non-hypertensive are: 500
The percentage of patients with hypertension is: 50.0%


## 8. Patients aged 65 and above

In [ ]:
older_patients = age >= 65
older_count = np.sum(older_patients)
percentage_65_above = (older_count / len(age)) * 100

print(f"The number of patients aged 65 and above is: {older_count}")
print(f"The percentage of patients aged 65 and above is: {percentage_65_above}%")

The number of patients aged 65 and above is: 183
The percentage of patients aged 65 and above is: 18.3%


## 9. Comorbidity: diabetes AND hypertension

`&` is NumPy's element-wise AND: it compares two boolean arrays position by position, and each result is `True` only where **both** arrays are `True` at that position.

In [ ]:
diabetic = diabetes == "Yes"
hypertensive = htn == "Yes"

both = diabetic & hypertensive
both_count = np.sum(both)
both_percentage = (both_count / len(diabetes)) * 100

print(f"Number of patients with diabetes and hypertension: {both_count}")
print(f"Percentage with both conditions: {both_percentage:.1f}%")

Number of patients with diabetes and hypertension: 106
Percentage with both conditions: 10.6%


## 10. Summary

| Metric | Value |
|---|---|
| Patients | 1,000 |
| Mean age | 51.6 years |
| Median age | 52 years |
| Mean BMI | 27.4 kg/m² |
| Mean systolic BP | 139.8 mmHg |
| Std. dev. systolic BP | 18.2 |
| Highest HbA1c | 9.1% |
| Lowest fasting glucose | 55.0 mg/dL |
| Diabetes prevalence | 15.1% (151 patients) |
| Hypertension prevalence | 50.0% (500 patients) |
| Aged 65+ | 18.3% (183 patients) |
| Diabetes + hypertension | 10.6% (106 patients) |

**Techniques practiced:** `np.mean`, `np.median`, `np.std`, `np.nanmean`, `np.nanmax`, `np.min`, boolean masking (`==`, `>=`), and combining conditions with `&`.